frontend/packages/kubevirt-plugin/src/components/modals/cpu-pinning-modal/cpu-pinning-modal.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { Checkbox, Modal, Title } from '@patternfly/react-core';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { withHandlePromise, HandlePromiseProps } from '@console/internal/components/utils';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { ModalFooter } from '../modal/modal-footer';
+import { getVMLikeModel } from '../../../selectors/vm';
+import { getCPUPinningPatch } from '../../../k8s/patches/vm/vm-cpu-patches';
+import { VMLikeEntityKind } from '../../../types';
+import './cpu-pinning-modal.scss';
+
+export const CpuPinningModal = withHandlePromise(
+  ({
+    vmLikeEntity,
+    isCPUPinned,
+    isOpen,
+    setOpen,
+    title = 'Devices & Resources',
+    handlePromise,
+    inProgress,
+    errorMessage,
+  }: CpuPinningModalProps) => {
+    const [isPinned, setIsPinned] = React.useState<boolean>(isCPUPinned);
```

Comment:
I agree @suomiy but I think @jelkosz felt like this is more advanced than flavor. @Ranelim have an effort this sprint to look at topology and how it could align with this approach.